pkg/connector/backfill.go

```diff
@@ -46,16 +54,47 @@ func (t *TelegramClient) getTakeoutID(ctx context.Context) (takeoutID int64, err
 			return 0, err
 		}
 
-		if t.stopTakeoutTimer != nil {
-			t.stopTakeoutTimer.Stop()
-		}
-		t.stopTakeoutTimer = time.AfterFunc(max(time.Hour, time.Duration(t.main.Bridge.Config.Backfill.Queue.BatchDelay*2)), sync.OnceFunc(func() { t.stopTakeout(ctx) }))
+		// Fetch all chats using takeout and enqueue them for syncing.
+		go t.takeoutDialogs(ctx, takeoutID)
```

Comment:
Dialogs should only be fully resynced if they haven't been fetched before, should probably store a flag in `UserLoginMetadata` for that (you can make it fancy so it continues from previous position if disrupted, but just a boolean works too)